R/matrixfit.R

```diff
@@ -1,791 +1,462 @@
-}
-
-## reference_time is a dummy variable here
-dmatrixChisqr.shifted <- function(par, t, y, M, T, parind, sign.vec, ov.sign.vec, deltat=1, reference_time=0) {
-  res <- rep(0., times=length(par))
-  z <- (y-ov.sign.vec*par[parind[,1]]*par[parind[,2]]*(exp(-par[1]*(t-deltat/2)) - sign.vec*exp(-par[1]*(T-(t-deltat/2)))))
-  zp <- -ov.sign.vec*par[parind[,1]]*par[parind[,2]]*(-(t-deltat/2)*exp(-par[1]*(t-deltat/2)) + (T-t+deltat/2)*sign.vec*exp(-par[1]*(T-(t-deltat/2))))
-  res[1] <- sum(zp %*% M %*% z + z %*% M %*% zp)
-  for(i in 2:length(par)) {
-    zp <- rep(0, length(z))
-    j <- which(parind[,1]==i)
-    zp[j] <- -ov.sign.vec[j]*par[parind[j,2]]*(exp(-par[1]*(t[j]-deltat/2)) - sign.vec[j]*exp(-par[1]*(T-(t[j]-deltat/2))))
-    res[i] <- sum(zp %*% M %*% z + z %*% M %*% zp)
-    zp <- rep(0, length(z))
-    j <- which(parind[,2]==i)
-    zp[j] <- -ov.sign.vec[j]*par[parind[j,1]]*(exp(-par[1]*(t[j]-deltat/2)) - sign.vec[j]*exp(-par[1]*(T-(t[j]-deltat/2))))
-    res[i] <- res[i] + sum(zp %*% M %*% z + z %*% M %*% zp)
-  }
-  return(res)
-}
-
-## reference_time is a dummy variable here
-matrixChi.shifted <- function(par, t, y, L, T, parind, sign.vec, ov.sign.vec, deltat=1, reference_time=0) {
-  z <- (y-ov.sign.vec*par[parind[,1]]*par[parind[,2]]*(exp(-par[1]*(t-deltat/2)) - sign.vec*exp(-par[1]*(T-(t-deltat/2)))))
-  return( L %*% z )
-}
-
-## reference_time is a dummy variable here
-dmatrixChi.shifted <- function(par, t, y, L, T, parind, sign.vec, ov.sign.vec, deltat=1, reference_time=0) {
-  zp <- -ov.sign.vec*par[parind[,1]]*par[parind[,2]]*(-(t-deltat/2)*exp(-par[1]*(t-deltat/2)) +(T-t+deltat/2)*sign.vec*exp(-par[1]*(T-(t-deltat/2))))
-  res <- L %*% zp
-  for(i in 2:length(par)) {
-    zp1 <- c(0)
-    j <- which(parind[,1]==i)
-    zp1[j] <- -ov.sign.vec[j]*par[parind[j,2]]*(exp(-par[1]*(t[j]-deltat/2)) - sign.vec[j]*exp(-par[1]*(T-(t[j]-deltat/2))))
-    zp2 <- c(0)
-    j <- which(parind[,2]==i)
-    zp2[j] <- -ov.sign.vec[j]*par[parind[j,1]]*(exp(-par[1]*(t[j]-deltat/2)) - sign.vec[j]*exp(-par[1]*(T-(t[j]-deltat/2))))
-    res <- c(res, L %*% zp1 + L %*% zp2)
-  }
-  return(res)
-}
-
-
-# the calling code must supply the correct three parameters 
-deriv.CExp <- function(par, t, T, sign) {
-  res <- array(0.,dim=c(length(par),length(t)))
-
-  res[1,] <- 0.5*par[2]*par[3]*(-t*exp(-par[1]*t) -(T-t)*sign*exp(-par[1]*(T-t)))
-  res[2,] <- 0.5*par[3]*(exp(-par[1]*t) + sign*exp(-par[1]*(T-t)))
-  res[3,] <- 0.5*par[2]*(exp(-par[1]*t) + sign*exp(-par[1]*(T-t)))
-
-  return(res)
-}
-
-deriv.CExp.shifted <- function(par, t, T, sign, deltat=1) {
-  res <- array(0.,dim=c(length(par),length(t)))
-  
-  res[1,] <- par[2]*par[3]*(-(t-deltat/2)*exp(-par[1]*(t-deltat/2)) + (T-(t-deltat/2))*sign*exp(-par[1]*(T-(t-deltat/2))))
-  res[2,] <- par[3]*(exp(-par[1]*(t-deltat/2)) - sign*exp(-par[1]*(T-(t-deltat/2))))
-  res[3,] <- par[2]*(exp(-par[1]*(t-deltat/2)) - sign*exp(-par[1]*(T-(t-deltat/2))))
-
-  return(res)
-}
+library(R6)
```

Comment:
is the usage of ```library()``` in a package legitimate for ```R6```? Weren't we supposed to use ```requireNameSpace```?

Reply:
Yes, because it is a compile time dependency at this point. We need to to declare the class, but we do not need it at runtime.